src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,48 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import static edu.kis.powp.jobs2d.features.DrawerFeature.getDrawerController;
+
+public class LineDrawerAdapter extends DrawPanelController implements Job2dDriver {
+    private int startX = 0, startY = 0;
+    private ILine line;
+
+    public LineDrawerAdapter() {
+        super();
+        line = LineFactory.getSpecialLine();
+    }
+
+
+    public void setTypeLine(String type){
+        if(type.equals("basic"))
+            line = LineFactory.getBasicLine();
+        else if(type.equals("special"))
+            line = LineFactory.getSpecialLine();
```

Comment:
OK, but coupling with LineFactory is not necessary (with e.g., constructor injection)